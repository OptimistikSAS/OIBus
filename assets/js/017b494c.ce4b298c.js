"use strict";(self.webpackChunkdoc_oibus=self.webpackChunkdoc_oibus||[]).push([[9615],{7972:e=>{e.exports=JSON.parse('{"categoryGeneratedIndex":{"title":"Engine","description":"Learn how to set up OIBus Engine.","slug":"/category/engine","permalink":"/docs/category/engine","sidebar":"guideSidebar","navigation":{"previous":{"title":"Migrate to v3","permalink":"/docs/guide/installation/migrate"},"next":{"title":"Engine settings","permalink":"/docs/guide/engine/engine-settings"}}}}')}}]);