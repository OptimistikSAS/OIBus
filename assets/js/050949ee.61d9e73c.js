"use strict";(self.webpackChunkdoc_oibus=self.webpackChunkdoc_oibus||[]).push([[3288],{1966:s=>{s.exports=JSON.parse('{"name":"docusaurus-plugin-content-docs","id":"default"}')}}]);