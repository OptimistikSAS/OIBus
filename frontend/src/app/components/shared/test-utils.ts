import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
