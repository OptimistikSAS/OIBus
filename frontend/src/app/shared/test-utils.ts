import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
