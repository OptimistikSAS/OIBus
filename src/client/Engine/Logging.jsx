import React from 'react'
import PropTypes from 'prop-types'
import { Row, Col } from 'reactstrap'
import { OIbInteger, OIbSelect, OIbText, OIbCheckBox, OIbTitle } from '../components/OIbForm'
import validation from './Engine.validation'

const Logging = ({ logParameters, onChange }) => (
  <>
    <OIbTitle label="Log Parameters">
      <>
        <p>OIBus can send logs to 3 different supports:</p>

        <ul>
          <li>The Console logs message on the terminal. It cannot be used when OIBus runs as a Windows service</li>
          <li>The File logs logs message on the chosen folder of the server</li>
          <ul>
            <li>filename: The filename of the logfile to write output to.</li>
            <li>
              maxsize: Max size in bytes of the logfile, if the size is exceeded then a new file is created,
              a counter will become a suffix of the log file.
            </li>
            <li>maxFiles: Limit the number of files created when the size of the logfile is exceeded.</li>
            <li>
              tailable: If true, log files will be rolled based on maxsize and maxfiles, but in ascending order. The filename will always have the
              most recent log lines. The larger the appended number, the older the log file. This option requires maxFiles to be set, or it will be
              ignored.
            </li>
          </ul>
          <li>
            The sqlite logs will be used to store logs on the chosen database of the server.
            This allows to see logs remotely using the Logs menu. The
            maximum size can be defined so older message will be deleted automatically.
          </li>
        </ul>

        <table>
          <thead>
            <tr>
              <th>Possible levels</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Error</td>
              <td>these logs needs investigation in a production system</td>
            </tr>
            <tr>
              <td>Warning</td>
              <td>unexpected behavior that will not impact significantly the system</td>
            </tr>
            <tr>
              <td>Info</td>
              <td>information useful to the administrator</td>
            </tr>
            <tr>
              <td>Debug</td>
              <td>informations useful only to debug an issue</td>
            </tr>
            <tr>
              <td>Silly</td>
              <td>informations useful to debug an issue but they will significantly impact performances</td>
            </tr>
          </tbody>
        </table>
        <br />
        <p>
          We recommend to use the info or warning levels in normal operations and to put the debug or the silly mode only to identify the origin of an
          issue.
        </p>
      </>
    </OIbTitle>
    <Row>
      <Col md={3}>
        <OIbSelect
          label="Console Level"
          name="engine.logParameters.consoleLevel"
          value={logParameters.consoleLevel}
          options={['silly', 'debug', 'info', 'warning', 'error']}
          defaultValue="info"
          help={<div>The level for the Console log</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={3}>
        <OIbSelect
          name="engine.logParameters.fileLevel"
          label="File Level"
          options={['silly', 'debug', 'info', 'warning', 'error']}
          defaultValue="info"
          value={logParameters.fileLevel}
          help={<div>The level for the file log</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={3}>
        <OIbSelect
          label="Sqlite Level"
          name="engine.logParameters.sqliteLevel"
          options={['silly', 'debug', 'info', 'warning', 'error']}
          defaultValue="info"
          value={logParameters.sqliteLevel}
          help={<div>The level for the sqlite log</div>}
          onChange={onChange}
        />
      </Col>
    </Row>
    <Row>
      <Col md={4}>
        <OIbText
          name="engine.logParameters.filename"
          label="Filename for the log file"
          value={logParameters.filename}
          valid={validation.engine.logParameters.filename}
          help={<div>The filename of the log file</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={2}>
        <OIbInteger
          name="engine.logParameters.maxsize"
          label="File Max size"
          value={logParameters.maxsize}
          defaultValue={100000}
          valid={validation.engine.logParameters.maxsize}
          help={<div>Maximum size of file logs (Bytes)</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={2}>
        <OIbInteger
          name="engine.logParameters.maxFiles"
          label="number of files"
          value={logParameters.maxFiles}
          defaultValue={5}
          valid={validation.engine.logParameters.maxFiles}
          help={<div>The number of journal files (rotating)</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={3}>
        <OIbCheckBox label="Tailable" name="engine.logParameters.tailable" value={logParameters.tailable} defaultValue onChange={onChange} />
      </Col>
    </Row>
    <Row>
      <Col md={4}>
        <OIbText
          name="engine.logParameters.sqliteFilename"
          label="Filename of sqlite db"
          value={logParameters.sqliteFilename}
          valid={validation.engine.logParameters.sqliteFilename}
          help={<div>The filename of the db file</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={2}>
        <OIbInteger
          name="engine.logParameters.sqliteMaxFileSize"
          label="Db Max size"
          value={logParameters.sqliteMaxFileSize}
          defaultValue={1000000}
          valid={validation.engine.logParameters.sqliteMaxFileSize}
          help={<div>Max File Size of the sqlite database (Byte)</div>}
          onChange={onChange}
        />
      </Col>
    </Row>
  </>
)

Logging.propTypes = {
  onChange: PropTypes.func.isRequired,
  logParameters: PropTypes.object.isRequired,
}
export default Logging
