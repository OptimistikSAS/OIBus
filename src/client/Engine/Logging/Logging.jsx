import React from 'react'
import PropTypes from 'prop-types'
import OIbTitle from '../../components/OIbForm/OIbTitle.jsx'

import ConsoleLogging from './ConsoleLogging.jsx'
import FileLogging from './FileLogging.jsx'
import SqliteLogging from './SqliteLogging.jsx'
import LokiLogging from './LokiLogging.jsx'

const Logging = ({ logParameters, onChange }) => (
  <div>
    <OIbTitle label="Logging parameters">
      <>
        <p>OIBus can send logs to 4 different supports:</p>

        <ul>
          <li>The Console logs message on the terminal. It cannot be used when OIBus runs as a Windows service</li>
          <li>The File logs logs message on the chosen folder of the server</li>
          <ul>
            <li>filename: The filename of the logfile to write output to.</li>
            <li>
              maxsize: Max size in bytes of the logfile, if the size is exceeded then a new file is created, a counter
              will become a suffix of the log file.
            </li>
            <li>maxFiles: Limit the number of files created when the size of the logfile is exceeded.</li>
            <li>
              tailable: If true, log files will be rolled based on maxsize and maxfiles, but in ascending order. The
              filename will always have the most recent log lines. The larger the appended number, the older the log
              file. This option requires maxFiles to be set, or it will be ignored.
            </li>
          </ul>
          <li>
            The sqlite logs will be used to store logs on the chosen database of the server. This allows to see logs
            remotely using the Logs menu. The maximum size can be defined so older message will be deleted
            automatically.
          </li>
          <li>
            The loki logs send the logs to a loki instance. This allows to access the logs remotely.
          </li>
        </ul>

        <table>
          <thead>
            <tr>
              <th>Levels</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Error</td>
              <td>These logs needs investigation in a production system</td>
            </tr>
            <tr>
              <td>Warning</td>
              <td>Unexpected behavior that will not impact significantly the system</td>
            </tr>
            <tr>
              <td>Info</td>
              <td>Information useful to the administrator</td>
            </tr>
            <tr>
              <td>Debug</td>
              <td>Information useful only to debug an issue</td>
            </tr>
            <tr>
              <td>Silly</td>
              <td>Information useful to debug an issue but they will significantly impact performances</td>
            </tr>
            <tr>
              <td>None</td>
              <td>This logging is not activated</td>
            </tr>
          </tbody>
        </table>
        <br />
        <p>
          We recommend to use the info or warning levels in normal operations and to put the debug or the silly mode
          only to identify the origin of an issue.
        </p>
      </>
    </OIbTitle>
    <ConsoleLogging logParameters={logParameters.consoleLog} onChange={onChange} />
    <FileLogging logParameters={logParameters.fileLog} onChange={onChange} />
    <SqliteLogging logParameters={logParameters.sqliteLog} onChange={onChange} />
    <LokiLogging logParameters={logParameters.lokiLog} onChange={onChange} />
  </div>
)

Logging.propTypes = {
  onChange: PropTypes.func.isRequired,
  logParameters: PropTypes.object.isRequired,
}
export default Logging
