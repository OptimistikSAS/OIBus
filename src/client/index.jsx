import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { Container } from 'reactstrap'

import './style/main.less'

import TopHeader from './TopHeader.jsx'
import Activation from './Activation/Activation.jsx'
import NotFound from './NotFound.jsx'
import South from './South/South.jsx'
import North from './North/North.jsx'
import Engine from './Engine/Engine.jsx'
import ConfigureApi from './North/ConfigureApi.jsx'
import ConfigureProtocol from './South/ConfigureProtocol.jsx'
import ConfigurePoints from './South/ConfigurePoints.jsx'
import Logs from './Logs/Logs.jsx'
import Health from './Health/Health.jsx'
import AlertContainer from './components/AlertContainer.jsx'
import { AlertProvider } from './context/AlertContext.jsx'
import { ConfigProvider } from './context/configContext.jsx'

const Main = () => (
  <Router>
    <>
      <ConfigProvider>
        <AlertProvider>
          <TopHeader />
          <Container className="oi-container-with-top-nav" fluid>
            <AlertContainer />
            <Switch>
              <Route exact path="/" component={Health} />
              <Route exact path="/engine" component={Engine} />
              <Route exact path="/south" component={South} />
              <Route exact path="/south/:dataSourceId" component={ConfigureProtocol} />
              <Route exact path="/south/:dataSourceId/points" component={ConfigurePoints} />
              <Route exact path="/north" component={North} />
              <Route exact path="/north/:applicationId" component={ConfigureApi} />
              <Route exact path="/log" component={Logs} />
              <Route exact path="/activation" component={Activation} />
              <Route component={NotFound} />
            </Switch>
          </Container>
        </AlertProvider>
      </ConfigProvider>
    </>
  </Router>
)

ReactDOM.render(<Main />, document.getElementById('root'))
