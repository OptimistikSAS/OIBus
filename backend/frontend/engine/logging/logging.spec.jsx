/**
 * @jest-environment jsdom
 */
import React from 'react'
import { act } from 'react-dom/test-utils'

import * as ReactDOMClient from 'react-dom/client'
import { testConfig } from '../../../tests/test-config.js'
import Logging from './logging.jsx'

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
let root
// eslint-disable-next-line no-undef
globalThis.IS_REACT_ACT_ENVIRONMENT = true
beforeEach(() => {
  container = document.createElement('div')
  root = ReactDOMClient.createRoot(container)
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
  root = null
})

describe('Logging', () => {
  test('check Logging', () => {
    act(() => {
      root.render(<Logging
        logParameters={testConfig.engine.logParameters}
        onChange={() => (1)}
      />)
    })
    expect(container).toMatchSnapshot()
  })
})
