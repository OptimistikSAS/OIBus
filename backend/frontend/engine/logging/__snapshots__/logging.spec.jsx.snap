// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logging check Logging 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <h5>
        Logging parameters
        <button
          class="util-button mt-1 btn btn-link"
          id="id"
        >
          <svg
            class="oi-help"
            fill="currentColor"
            height="12"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
            />
          </svg>
        </button>
      </h5>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="collapse"
        toggler="id"
      >
        <div
          class="row"
          style="margin-bottom: 15px;"
        >
          <div
            class="col"
          >
            <p>
              OIBus can send logs to 4 different supports:
            </p>
            <ul>
              <li>
                The Console logs messages on the terminal. It cannot be used when OIBus runs as a Windows service
              </li>
              <li>
                The File logs writes messages on the chosen folder of the OIBus machine
              </li>
              <ul>
                <li>
                  File name: The file name of the logfile to write messages to.
                </li>
                <li>
                  Max size: Max size in bytes of the logfile, if the size is exceeded then a new file is createdIn this case, a counter will be added as a suffix of the logfile name.
                </li>
                <li>
                  Max files: Limit the number of files created by the logger.
                </li>
              </ul>
              <li>
                The sqlite logs will be used to store logs on the chosen database of the OIBus machine. This allows to see logs using the Logs menu. The maximum size can be defined so older message will be deleted automatically.
              </li>
              <li>
                The loki logs send the logs to a loki instance. This allows to access the logs remotely. If no token address is provided, the logs can be sent using basic auth (if username/password is provided) or with no auth. If the token address is provided, the token is retrieved using basic auth (username/password) and the token will be used in the Authorization header field as a Bearer token.
              </li>
            </ul>
            <table>
              <thead>
                <tr>
                  <th>
                    Levels
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Silent
                  </td>
                  <td>
                    This logging is not activated
                  </td>
                </tr>
                <tr>
                  <td>
                    Error
                  </td>
                  <td>
                    These logs needs investigation in a production system
                  </td>
                </tr>
                <tr>
                  <td>
                    Warning
                  </td>
                  <td>
                    Unexpected behavior that will not impact significantly the system
                  </td>
                </tr>
                <tr>
                  <td>
                    Info
                  </td>
                  <td>
                    Information useful to the administrator
                  </td>
                </tr>
                <tr>
                  <td>
                    Debug
                  </td>
                  <td>
                    Information useful only to debug an issue
                  </td>
                </tr>
                <tr>
                  <td>
                    Trace
                  </td>
                  <td>
                    Information useful to debug an issue but they will significantly impact performances
                  </td>
                </tr>
              </tbody>
            </table>
            <br />
            <p>
              We recommend to use the info or warning levels in normal operations and to put the debug or the trace mode only to identify the origin of an issue.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h6>
        Console
      </h6>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.consoleLog.level"
            >
              Level
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-select"
              id="engine.logParameters.consoleLog.level"
              name="engine.logParameters.consoleLog.level"
            >
              <option
                value="trace"
              >
                trace
              </option>
              <option
                value="debug"
              >
                debug
              </option>
              <option
                value="info"
              >
                info
              </option>
              <option
                value="warning"
              >
                warning
              </option>
              <option
                value="error"
              >
                error
              </option>
              <option
                value="silent"
              >
                silent
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The level for the Console log
              </div>
            </small>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h6>
        File
      </h6>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.fileLog.level"
            >
              Level
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-select"
              id="engine.logParameters.fileLog.level"
              name="engine.logParameters.fileLog.level"
            >
              <option
                value="trace"
              >
                trace
              </option>
              <option
                value="debug"
              >
                debug
              </option>
              <option
                value="info"
              >
                info
              </option>
              <option
                value="warning"
              >
                warning
              </option>
              <option
                value="error"
              >
                error
              </option>
              <option
                value="silent"
              >
                silent
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The level for the File log
              </div>
            </small>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-2"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.fileLog.maxSize"
            >
              Max file size
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.fileLog.maxSize"
              name="engine.logParameters.fileLog.maxSize"
              type="integer"
              value="1000000"
            />
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                Maximum size of the log files (MB)
              </div>
            </small>
          </div>
        </div>
        <div
          class="col-md-2"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.fileLog.numberOfFiles"
            >
              Number of files
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.fileLog.numberOfFiles"
              name="engine.logParameters.fileLog.numberOfFiles"
              type="integer"
              value="5"
            />
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The number of log files (rotating)
              </div>
            </small>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h6>
        SQLite
      </h6>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.sqliteLog.level"
            >
              Level
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-select"
              id="engine.logParameters.sqliteLog.level"
              name="engine.logParameters.sqliteLog.level"
            >
              <option
                value="trace"
              >
                trace
              </option>
              <option
                value="debug"
              >
                debug
              </option>
              <option
                value="info"
              >
                info
              </option>
              <option
                value="warning"
              >
                warning
              </option>
              <option
                value="error"
              >
                error
              </option>
              <option
                value="silent"
              >
                silent
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The level for the Sqlite log
              </div>
            </small>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-2"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.sqliteLog.maxNumberOfLogs"
            >
              Max number of logs
            </label>
            <input
              aria-invalid="true"
              class="oi-form-input is-invalid form-control"
              id="engine.logParameters.sqliteLog.maxNumberOfLogs"
              name="engine.logParameters.sqliteLog.maxNumberOfLogs"
              type="integer"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Value must be greater than 100000
            </div>
            <small
              class="form-text text-muted"
            >
              <div>
                Maximum number of logs in the SQLite database
              </div>
            </small>
          </div>
        </div>
      </div>
    </div>
    <div>
      <h6>
        Loki
      </h6>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.level"
            >
              Level
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-select"
              id="engine.logParameters.lokiLog.level"
              name="engine.logParameters.lokiLog.level"
            >
              <option
                value="trace"
              >
                trace
              </option>
              <option
                value="debug"
              >
                debug
              </option>
              <option
                value="info"
              >
                info
              </option>
              <option
                value="warning"
              >
                warning
              </option>
              <option
                value="error"
              >
                error
              </option>
              <option
                value="silent"
              >
                silent
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The level for the Loki log
              </div>
            </small>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.lokiAddress"
            >
              Host
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.lokiAddress"
              name="engine.logParameters.lokiLog.lokiAddress"
              type="text"
              value="localhost:3100"
            />
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The host name of the Loki instance
              </div>
            </small>
          </div>
        </div>
        <div
          class="col-md-2"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.interval"
            >
              Interval (s)
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.interval"
              name="engine.logParameters.lokiLog.interval"
              type="integer"
              value="60"
            />
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                Interval between batch of logs (in s)
              </div>
            </small>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.tokenAddress"
            >
              Token address
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.tokenAddress"
              name="engine.logParameters.lokiLog.tokenAddress"
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            />
            <small
              class="form-text text-muted"
            >
              <div>
                The address of the token provider (need username and password)
              </div>
            </small>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.username"
            >
              Username
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="engine.logParameters.lokiLog.username"
              name="engine.logParameters.lokiLog.username"
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            />
          </div>
        </div>
        <div
          class="col-md-3"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="engine.logParameters.lokiLog.password"
            >
              Password
            </label>
            <div
              class="input-group"
            >
              <input
                aria-invalid="false"
                class="oi-form-input form-control"
                id="engine.logParameters.lokiLog.password"
                name="engine.logParameters.lokiLog.password"
                type="password"
                value=""
              />
            </div>
            <div
              class="invalid-feedback"
              style="display: inline;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
